app/controllers/chargeback_controller.rb

```diff
@@ -179,10 +185,21 @@ def cb_rate_edit
         end
       else      # for save button
         cb_rate_set_record_vars
+        @sb[:rate].chargeback_rate_details.each_with_index do |_detail, i|
+          @sb[:rate_details][i].chargeback_tiers = @sb[:tiers][i]
+        end
+
+        # Detect errors saving tiers
+        tier_error = false
+        @sb[:tiers].each { |tiers| tiers.each { |tier| tier_error = true if tier.valid? == false } }
+
         # Detect errors saving rate details
         rate_detail_error = false
-        @sb[:rate_details].each { |detail| rate_detail_error = true if detail.save == false }
-        if rate_detail_error == false && @sb[:rate].save
+        @sb[:rate_details].each { |detail| rate_detail_error = true if detail.valid? == false }
+        if rate_detail_error == false && tier_error == false && @sb[:rate].save
```

Comment:
Can we instead use a local variable that's positive instead of negative?

I would much rather read `if rate_detail_valid && tier_valid && @sb[:rate].save`

Perhaps this code could be simplified by inverting the logic above to deal with the positive case like...


`rate_detail_valid = @sb[:rate_details].all? { |detail| detail.valid?}`